crates/energy/benches/nearby.rs

```diff
@@ -0,0 +1,142 @@
+use std::fs::File;
+use std::io::{BufRead, BufReader};
+use std::path::PathBuf;
+
+use bevy::ecs::schedule::ScheduleLabel;
+use bevy::prelude::{
+    App, Changed, Component, Entity, Query, Res, ResMut, Schedule, Time, Transform, Update, Vec2,
+    World,
+};
+use bevy::time::TimePlugin;
+use criterion::{criterion_group, criterion_main, Criterion};
+use de_energy::{update_nearby_recv, EnergyReceiver, NearbyUnits};
+use de_index::{EntityIndex, LocalCollider};
+use de_objects::ObjectCollider;
+use parry3d::math::{Isometry, Vector};
+use parry3d::shape::{Cuboid, TriMesh};
+
+const MAP_SIZE: f32 = 2000.;
+const DISTANCE_FROM_MAP_EDGE: f32 = 100.;
+
+#[derive(ScheduleLabel, Clone, Debug, PartialEq, Eq, Hash)]
+pub struct UpdateUnits;
+
+#[derive(Component)]
+struct UnitNumber(u32);
+
+fn update_index(
+    mut index: ResMut<EntityIndex>,
+    moved: Query<(Entity, &Transform), Changed<Transform>>,
+) {
+    for (entity, transform) in moved.iter() {
+        let position = Isometry::new(
+            transform.translation.into(),
+            transform.rotation.to_scaled_axis().into(),
+        );
+        index.update(entity, position);
+    }
+}
+
+fn load_points(number: u32) -> Vec<Vec2> {
+    let mut points_path: PathBuf = env!("CARGO_MANIFEST_DIR").into();
+    points_path.push("test_data");
+    points_path.push(format!("{number}-points.txt"));
+    let reader = BufReader::new(File::open(points_path).unwrap());
+
+    let mut points = Vec::with_capacity(number as usize);
+    for line in reader.lines() {
+        let line = line.unwrap();
+        let mut numbers = line.split_whitespace();
+        let x: f32 = numbers.next().unwrap().parse().unwrap();
+        let y: f32 = numbers.next().unwrap().parse().unwrap();
+        points.push((MAP_SIZE - DISTANCE_FROM_MAP_EDGE) * Vec2::new(x, y));
+    }
+    points
+}
+
+fn init_world_with_entities_moving(world: &mut World, num_entities: u32) {
+    let mut index = EntityIndex::new();
+    let points = load_points(num_entities);
+
+    for (i, point) in points.into_iter().enumerate() {
+        let x: f32 = point.x;
+        let y: f32 = point.y;
+
+        let collider = LocalCollider::new(
+            ObjectCollider::from(TriMesh::from(Cuboid::new(Vector::new(3., 3., 4.)))),
+            Isometry::new(Vector::new(x, 0., y), Vector::identity()),
+        );
+
+        let entity = world
+            .spawn((
+                Transform::from_xyz(x, y, 0.0),
```

Comment:
Should be `x, 0., -y`. But better: `let point_msl = point.to_msl()` (implemented by `ToAltitude`) and use `point_msl` for both the isometry and this transform (`Transform::from_translation(point_msl)`).